### PR TITLE
Add sort command

### PR DESCRIPTION
**Pull request by yitong241**
Add sort command

### CODE REVIEW

File: src/main/java/seedu/address/model/patient/Patient.java [hunk 1]
```diff
@@ -168,6 +179,71 @@ public boolean isSameNric(Patient otherPatient) {
...
+        return this.getNameString().compareTo(patient.getNameString());
+    }
+
+
+    /**
+     * Returns compared result between {@code this} and the given {@code patient} by Name in descending order.
+     * Returns positive integer if {@code this} should be placed after, 0 if same, and negative if before.
+     */
+    public int compareToByNameDesc(Patient patient) {
+        return -1 * this.getNameString().compareTo(patient.getNameString());
```
Comment (KSunil2001) [inline src/main/java/seedu/address/model/patient/Patient.java:196]: You can store -1 as a variable to indicate its purpose. Same for the other uses of -1 below this code.

File: src/test/java/seedu/address/model/AddressBookTest.java [hunk 2]
```diff
@@ -7,10 +7,7 @@
 import static seedu.address.testutil.TypicalPatients.ALEX;
 import static seedu.address.testutil.TypicalPatients.getTypicalAddressBook;
 
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.List;
+import java.util.*;
```
Comment (KSunil2001) [inline src/test/java/seedu/address/model/AddressBookTest.java:10]: Avoid using * for imports to improve code style

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        requireNonNull(model);
+        model.sortPatients(this.comparator);
+        return new CommandResult(MESSAGE_SUCCESS);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return this == other // short circuit if same object
+                || (other instanceof SortCommand // instanceof handles null
+                && this.comparator.equals(((SortCommand) other).comparator)); // state check
```
Comment (KSunil2001) [inline src/main/java/seedu/address/logic/commands/SortCommand.java:122]: Maybe can apply some abstraction here to improve the code style 🤔

**Review by KSunil2001 — COMMENTED** [review]
Just some minor improvements to be made. Other than that, LGTM.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -126,4 +129,41 @@ public static Discharge parseDischarge(String discharge) throws ParseException {
...
+        case "STATUS":
+            return Field.STATUS;
+        case "WARD":
+            return Field.WARD;
+        case "DISCHARGE":
+            return Field.DISCHARGE;
+        default:
+            throw new ParseException(SortCommand.MESSAGE_UNKNOWN_TYPE_KEYWORD);
+        }
+    }
```
Comment (jeraldkiew) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:168]: Should use Enum.valueOf(String) to convert string to enum 👍🏻 

https://stackoverflow.com/questions/604424/how-to-get-an-enum-value-from-a-string-value-in-java

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        WARD
+    };
+
+    /**
+     * {@code ORDER} specifies what possible order {@code SortCommand} can accept.
+     */
+    public static enum Order {
+        ASC,
+        DESC
+    };
```
Comment (jeraldkiew) [inline src/main/java/seedu/address/logic/commands/SortCommand.java:65]: Really nice use of enums 👍🏻

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        case WARD:
+            if (order.equals(Order.ASC)) {
+                return Patient::compareToByWardAsc;
+            } else {
+                return Patient::compareToByWardDesc;
+            }
+        default:
+            // default sorting order is by Name Asc
+            return Patient::compareToByNameAsc;
+        }
```
Comment (jeraldkiew) [inline src/main/java/seedu/address/logic/commands/SortCommand.java:108]: feel like this should be abstracted to the enum

**Review by jeraldkiew — CHANGES_REQUESTED** [review]

**Review by jeraldkiew — APPROVED** [review]
LGTM, great work and quick changes Yitong! 👍🏻

**Pull request merged**